### PR TITLE
Tweaks

### PR DESCRIPTION
**Pull request by hadley**
Explained in individual comments

### CODE REVIEW

File: _frame.R [hunk 1]
```diff
@@ -1,10 +1,11 @@
 text_to_frame <- function(text, file, width = 1920, height = 360,
   bg = "#F8F8F8", fg = "#404040", ...) {
 
-  par(..., bg = bg)
-  png(file = file, width = width, height = height, bg = bg,
+  ragg::agg_png(filename = file, width = width, height = height, bg = bg,
```
Comment (hadley) [inline _frame.R:4]: This should be a little faster and guarantees the same rendering regardless of OS.

I double-checked and the CRAN version doesn't help with the Chinese rendering problem

File: _frame.R [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
     pointsize = 12)
-  par(mar = c(0,0,0,0))
+  on.exit(dev.off())
+
+  par(bg = bg, mar = c(0,0,0,0), ...)
```
Comment (hadley) [inline _frame.R:8]: I'm pretty sure setting `par()` before opening the graphics device didn't do anything.

File: generate.R [hunk 3]
```diff
@@ -57,7 +56,7 @@ generate_video <- function(srt_file, duration = "00:00", overwrite = "auto") {
     text_to_frame(df$text[i], file.path(captions_dir, paste0(df$n[i], ".png")))
   }
 
-  fps <- 24
+  fps <- 1
```
Comment (hadley) [inline generate.R:59]: I made this 1 to speed up testing; I'm not sure if it makes an appreciable difference to the output or not.

File: renv.lock [hunk 4]
```diff
@@ -4,7 +4,7 @@
     "Repositories": [
       {
         "Name": "CRAN",
-        "URL": "https://demo.rstudiopm.com/cran/latest"
+        "URL": "https://cloud.r-project.org"
```
Comment (hadley) [inline renv.lock:7]: This ensures that we get mac binaries; I have no idea what impact this has on long term reproducibility

File: generate.R [hunk 5]
```diff
@@ -82,18 +81,14 @@ generate_video <- function(srt_file, duration = "00:00", overwrite = "auto") {
...
-      "-y",                                         # Don't prompt for overwrite
-      "-r", "24",                                   # Framerate
-      "-s", "1920x360",                             # Resolution
-      "-i", file.path(frames_dir, "frame%08d.png"), # Input file
-      "-vcodec", "libx264", "-crf", "25",           # Set codec and parameter
-      "-pix_fmt", "yuv420p",                        # Output pixel format
-      output_file
-    ), echo_cmd = FALSE, echo = TRUE)
+    av::av_encode_video(
+      input = dir(frames_dir, full.names = TRUE),
```
Comment (hadley) [inline generate.R:87]: The other values appear to be derived correctly by default

**Comment by jcheng5** [comment]
Thanks! Let me test on Linux then I’ll merge

**Comment by hadley** [comment]
I think the change of repo will make linux requiring building from source 😞  Maybe you can provide both rspm and CRAN repos?

**Pull request merged**